app/containers/Forms/FuelField.tsx

```diff
@@ -29,13 +88,37 @@ export const FuelFieldComponent: React.FunctionComponent<Props> = (props) => {
     });
   };
 
+  const handleEmissionCategoryChange = (fuel) => {
+    const { emissionCategoryRowId, fuelRowId } = fuel;
+    const newFuelIds =
+      emissionCategoryRowId && fuelRowId
+        ? getFuelIdsByEmissionCategoryRowId(emissionCategoryRowId)
+        : null;
+    if (newFuelIds && newFuelIds.includes(fuelRowId)) onChange(fuel);
+    else
+      onChange({
+        ...fuel,
+        fuelRowId: undefined,
+        quantity: "",
+        fuelUnits: undefined,
+      });
+  };
+
```

Comment:
Same thing here, Just a comment describing the flow here where this function clears fuel data if the emission category is changed to one that does not include that fuel